src/components/views/Presets.svelte

```diff
@@ -1,3 +1,102 @@
-<div class="px-4">
-  Presets go here.
-</div>
\ No newline at end of file
+<script lang="ts">
+  import { patchStore, presetsStore } from '../../stores'
+  import type { Patch } from '$lib/patch'
+  import AddPreset from '$components/Presets/AddPreset.svelte'
+  import EditPreset from '$components/Presets/EditPreset.svelte'
+  import PresetInfo from '$components/Presets/PresetInfo.svelte'
+
+  let adding = false
+  let editing = false
+  let presets = $presetsStore.presets
+  $: selectedPreset = $presetsStore.presets.find(({ id }) => id === $presetsStore.selectedPatch) as Patch
+
+  // Update the selectedCategory and load the associated presets column
+  const handleCategoryClick = (category: string): void => {
+    presetsStore.update((state) => ({
+      ...state,
+      selectedCategory: category,
+    }))
+
+    switch (category) {
+      case $presetsStore.categories[0]:
+        presets = $presetsStore.presets
+        break
+
+      case $presetsStore.categories[1]:
+        presets = $presetsStore.presets.filter((preset) => preset.favorite)
+        break
+
+      default:
+        presets = $presetsStore.presets.filter((preset) => preset.tags.includes(category))
+        break
+    }
+
+    adding = false
+    editing = false
+  }
+
+  // Update the selectedPatch and load it into the patchStore
+  const handlePresetClick = (id: string) => {
+    presetsStore.update((state) => ({
+      ...state,
+      selectedPatch: id,
+    }))
+
+    selectedPreset = $presetsStore.presets.find(({ id }) => id === $presetsStore.selectedPatch) as Patch
```

Comment:
For some reason I haven't yet figured out, if I didn't look up the `selectedPreset` again here the `patchStore` was being updated with the previously selected patch, instead of the newly selected on. The strange thing is the `PresetInfo` component was getting the correct one passed into it even though they're using the same variable 🤔. I'll investigate that more tomorrow